packages/flutter/lib/src/material/input.dart

```diff
@@ -79,12 +188,29 @@ class _InputState extends State<Input> {
 
   GlobalKey get focusKey => config.key is GlobalKey ? config.key : _rawInputLineKey;
 
+  // Optional state to retain if we are inside a Form widget.
+  _FormFieldData _formData = null;
+
```

Comment:
if you don't like `_FormFieldData`, maybe `_FormFieldValue`? By convention, if the class ends with `State` that means it inherits from `State<T>`.